### DISCUSSION TITLE
Welcome to MQTTX Discussions!

### DISCUSSION THREAD

**Discussion by Rory-Z** [emqx/MQTTX#399] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by sbasheerjani** [comment]
Hello,
I am Basheer. I am trying to use MQTTx tool to connect to a MQTT Broker within the company. We use TLS v1.2 and self signed certificate. I am unable to connect to our MQTT broker and am getting below error. The connection works with other tool but not with MQTTx. Any suggestions/pointers would be much appreciated.
Error:
[2021-07-01 23:14:03] [ERROR] Connect fail, MQTT.js onError trigger, Error: Client network socket disconnected before secure TLS connection was established

**Reply by ysfscream** [reply]
Hi, Basheer. Thanks for using it. The verification method of MQTTX is stricter than other MQTT tools. If you are using a self-signed certificate, I suggest you refer to this blog:

https://www.emqx.io/blog/emqx-server-ssl-tls-secure-connection-configuration-guide

**Comment by nazran891** [comment]
Hello, just want to ask something. Can I install MQTTX inside my raspberry pi 3 B+?

**Reply by ysfscream** [reply]
YES, it should be possible, maybe you cloud to give it a try. 😃
